drv/i2c-devices/src/lib.rs

```diff
@@ -105,6 +150,40 @@ macro_rules! pmbus_write {
     }};
 }
 
+macro_rules! pmbus_rail_write {
+    ($device:expr, $rail:expr, $dev:ident::$cmd:ident, $data:expr) => {{
+        let rpayload = [$dev::PAGE::CommandData::code(), $rail];
+
+        let mut payload = [0u8; $dev::$cmd::CommandData::len() + 1];
+        payload[0] = $dev::$cmd::CommandData::code();
+        $data.to_slice(&mut payload[1..]);
+
+        match $device.write_write(&rpayload, &payload) {
+            Err(code) => Err(Error::BadWrite {
+                cmd: $dev::$cmd::CommandData::code(),
+                code,
+            }),
+            Ok(_) => Ok(()),
+        }
```

Comment:
I spent too much time trying to figure out way to remove this duplication:
```suggestion
        use $dev::{$cmd, PAGE};
        pmbus_rail_write!($device, $rail, $cmd, $data)
```

(the double `{{...}}` already make this is a local scope, so the `use` doesn't escape)

Reply:
That is a huge improvement!